dashboard/app/controllers/i18n_controller.rb

```diff
@@ -0,0 +1,29 @@
+class I18nController < ApplicationController
+  # The max number of i18n string keys which can be recorded in one request
+  I18N_KEY_COUNT_LIMIT = 500
+
+  # POST /i18n/track_string_usage
+  # Records the given i18n string usage information
+  # {
+  #   'string_keys': ['stringkey1', ...],
+  #   'url': 'https://code.org/some/page',
+  #   'source': 'maze',
+  # }
+  # string_keys [Array[String]] The i18n string keys which were used and need to be recorded.
+  # url [String] The URL where the i18n strings were used.
+  # source [String] Context for where the i18n strings came from.
+  def track_string_usage
+    string_keys = params.require(:string_keys)
+    url = params.require(:url)
+    source = params.require(:source)
+
+    # Limit the number of strings we will handle in one call.
+    return render status: :bad_request, json: {error: 'Too many strings.'}  if string_keys.count > I18N_KEY_COUNT_LIMIT
```

Comment:
Given that we're going to look into some kind of caching/batching to reduce firehose costs, is this limit still going to be necessary?

Reply:
This limit is more for our own protection against attacks where users manually make requests against the API will thousands/millions of items. It doesn't protect us from receiving the data, but it does prevent us from trying to doing something with it and holding up a single thread for a long time. I will update the documentation with that reasoning.